### PR TITLE
Individual participant refactoring

### PR DESCRIPTION
**Pull request by gostrobrod**
Reduce number of dependencies on IndividualParticipant. 
Move logic of Registration into Registration class. 
Make tests passing.

### CODE REVIEW

**Review by vteague — CHANGES_REQUESTED** [review]
This is much more logically structured, thankyou @gostrobrod.

I tested it through some account-creation flows and I did find one problem. Just to summarise what I tested:

- When I’m not already registered, and I try to ask a question and am directed through the ‘make an account’ -> ‘find your electorates’ -> ‘complete registration’ sequence, everything seems to work fine. Good.

- When I'm already registered, registering as an MP seems to work fine. Good.

- When I’m already registered, and I go to the Account page to update my details:
* changing my display_name works fine. Good.
* changing my state works fine. Good.
* changing my electorates doesn’t seem to work: it doesn’t get updated on the server. Bad.
 (The display also doesn’t update, but that’s an old problem - see issue #102  - don’t worry about that for now.) 
* However, when I close the app and re-open it, the new electorates are shown (which is good). They seem to be properly saved in preferences.

So I think the problem is that the logic for deciding whether to send updated electorates to the server is wrong. We have a data structure in the RegistrationViewModel called _oldElectorates - this is supposed to hold the prior electorates so that we can test whether the current ones are identical (and, if not, update). However, it seems to be being set to equal the new electorates. I suspect this is a consequence of building a new RegistrationViewModel. At the moment, when I debug 

- Line  283 of the RegistrationViewModel sets _oldElectorates to point to the same place as the new/current electorates.
- Line 39 sets it to be a new list with the contents of the new/current electorates (which would be right except by that stage it has the newly updated ones).

In other words, we want to make sure that _oldElectorates is set to be (a new list with the contents of) the current electorates, and then not reset when the person updates their electorates.

**Comment by vteague** [comment]
Or solve the problem any way that seems (more) logical to you, of course.

**Review by vteague — APPROVED** [review]
Terrific, thankyou.
